### PR TITLE
Added on-change support for bsdate and placeholder support for editable-select

### LINKED ISSUE #461: e-ng-change not working for bsdate element
Hi,

Please check out my plunker -> http://plnkr.co/edit/KzAwC1W0S7SZaG0wghl0

I've got a table form which contains a datepicker column. I want to utilize the e-ng-change method to do some work every time the date changes which works for basic text elements but not bsdate elements. I keep getting the below error when going to edit the table form:

> VM1830 angular.min.js:116 Error: [$compile:ctreq] http://errors.angularjs.org/1.5.3/$compile/ctreq?p0=ngModel&p1=ngChange
>     at Error (native)
>     at https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:6:416
>     at jb (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:70:406)
>     at n (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:65:259)
>     at g (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:58:140)
>     at g (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:58:157)
>     at g (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:58:157)
>     at https://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular.min.js:57:281
>     at EditableController.self.show (http://vitalets.github.io/angular-xeditable/dist/js/xeditable.js:716:30)
>     at http://vitalets.github.io/angular-xeditable/dist/js/xeditable.js:1174:26

I've tried adding e-ng-model but still to no avail. Any guidance would be greatly appreciated.

### PR DESCRIPTION
**Pull request by ckosloski**
- Added on-change support for bsdate
- Added placeholder support for editable-select
- Added tests for new functionality
- This closes #461, #423, #223, #429

### CODE REVIEW

File: src/js/directives/bsdate.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ angular.module('xeditable').directive('editableBsdate', ['editableDirectiveFacto
                  **/
                 this.parent.render.call(this);
 
-                var inputDatePicker = angular.element('<input type="text" class="form-control" ng-model="$data"/>');
+                var inputDatePicker = angular.element('<input type="text" class="form-control" data-ng-model="$data"/>');
```
Comment (eugef) [inline src/js/directives/bsdate.js]: Why did you change `ng-model` to `data-ng-model`? 
I don't think it is necessary.

File: docs/demos/dev-select/view.html [hunk 2]
```diff
@@ -1,5 +1,15 @@
 <div ng-controller="DevSelectCtrl">
-  <a href="#" editable-select="user.status" e-multiple e-ng-options="s.value as s.text for s in statuses">
```
Comment (eugef) [inline docs/demos/dev-select/view.html]: Why did you prefix all angular attributes names with `data`prefix? 
I don't think it is necessary.

Comment (ckosloski) [inline docs/demos/dev-select/view.html]: Yes, it's not necessary, it's a habit of mine.  I can remove them if you like.

Angular does list it in their best practices section on directives:
https://docs.angularjs.org/guide/directive

An overview of the data attribute
https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_data_attributes

Comment (ianellis90) [inline docs/demos/dev-select/view.html]: Thanks for the work @ckosloski. When will this be merged I could really do with this functionality?

**Comment by eugef** [comment]
@ckosloski, good job!
